### PR TITLE
Do not overwrite the SSH_DOMAIN when installing

### LINKED ISSUE #19453: Clone instructions do not respect `SSH_DOMAIN` variable when set as an environment variable
### Description

I'm running with the docker image and have the following set as environment variables. The idea being to have a separate domain for ssh cloning of repositories as the primary domain goes via Cloudflare proxy (and therefore blocks the SSH protocol).

```yaml
# file: docker-compose.yaml
gitea:
    image: gitea/gitea
    container_name: gitea
    hostname: git.mydomain.tld
    restart: always
    environment:
      - USER_UID=1000
      - USER_GID=1000
      - DB_TYPE=postgres
      - DB_HOST=db:5432
      - DB_NAME=gitea
      - DB_USER=gitea
      - DB_PASSWD=gitea
      # Customise
      - GITEA____APP_NAME="My Gitea Instance"
      - GITEA__server__DOMAIN="git.mydomain.tld"
      - GITEA__server__ROOT_URL="https://git.mydomain.tld"
      - GITEA__server__SSH_DOMAIN="clone.mydomain.tld"
    networks:
      - gitea
    volumes:
      - ./data/gitea:/data
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    depends_on:
      - db
      - redis
      - minio
```

On creating a new repository however I'm getting presenting with a SSH clone url which reflects the `DOMAIN` as opposed to the `SSH_DOMAIN`.

On investigation it seems as though this environment variable has not been respected and passed through to the underlying `app.ini`. If login into the container and inspect the file it shows the following:

```ini
# file: /data/gitea/conf/app.ini
...
[server]
APP_DATA_PATH    = /data/gitea
DOMAIN           = git.mydomain.tld
SSH_DOMAIN       = git.mydomain.tld
...
```

If I manually edit this file inside the container to adjust the `SSH_DOMAIN` variable and restart it then the setting is respected.

### Gitea Version

1.16.5 built with GNU Make 4.3, go1.18 : bindata, timetzdata, sqlite, sqlite_unlock_notify

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Gitea is running with docker and docker-compose using `image: gitea/gitea`

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by wxiaoguang**
Like

* #25203

Close #19453

* #19453

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: routers/install/install.go [hunk 1]
```diff
@@ -390,7 +390,7 @@ func SubmitInstall(ctx *context.Context) {
 	cfg.Section("database").Key("LOG_SQL").SetValue("false") // LOG_SQL is rarely helpful
 
 	cfg.Section("repository").Key("ROOT").SetValue(form.RepoRootPath)
-	cfg.Section("server").Key("SSH_DOMAIN").SetValue(form.Domain)
+	cfg.Section("server").Key("SSH_DOMAIN").MustString(form.Domain)
```
Comment (delvh) [inline routers/install/install.go:393]: Are we really going to convert each `SetValue` to `MustString` in a separate PR now?

**Review by delvh — APPROVED** [review]

File: routers/install/install.go (hunk 1 above)
Comment (wxiaoguang) [reply routers/install/install.go:393]: Do you have a better idea for this PR?

----

Put into draft, let's think more about it ....

Comment (delvh) [reply routers/install/install.go:393]: That wasn't meant as disapproval, I've simply noticed a tendency lately to replace `SetValue` in this method with `MustString`.
I'm wondering how many other settings we will convert as well in subsequent PRs.

**Comment by wxiaoguang** [comment]
I have a clear idea about this problem now.

 Make "install page" respect environment config #25648

**Pull request closed without merging**